Add development documentation page on documentation testing. (#607)
GeodynamicWorldBuilder/WorldBuilder: doc/sphinx/developer_manual/contributing_to_the_documentation/documentation_testing.md

Documentation testing
=====================

> Making sure that the examples in the documentation actually work is important to make sure potential and new users have a good experience. For that reason, all files in `doc/sphinx/_static/gwb_input_files/` which both have a `.wb` and `.grid` file, are automatically tested in every pull request. They are only tested to see if any error occur, not if the output is correct or has changed.  

Reviewer: ```suggestion
Making sure that the examples in the documentation are working is important to make sure users can rely on the documentation. For that reason, all files in `doc/sphinx/_static/gwb_input_files/` which both have a `.wb` and `.grid` file, are automatically tested in every pull request. They are only tested to see if any error occurs, not if the output is correct or has changed.
```
